electricitymap/contrib/lib/models/events.py

```diff
@@ -208,6 +253,59 @@ def to_dict(self) -> Dict[str, Any]:
         pass
 
 
+class AggregatableEvent(Event):
+    """
+    An abstract class representing all types of electricity events that can be aggregated.
+    """
+
+    @staticmethod
+    def _unique_zone_key(df_view: pd.DataFrame) -> ZoneKey:
+        zone_key = df_view["zoneKey"].unique()
+        if len(zone_key) > 1:
+            raise ValueError(f"Cannot merge events from different zones: {zone_key}")
+        return zone_key[0]
```

Comment:
Just as an edge case, we have a wrapping parser for the corsica and sardina exchange that merge DC and AC exchanges. Due to how the ENTSO-E parser was structured these needed to have different ZoneKeys.

Reply:
Not really sure I see what you mean, could you point me to the place this is done ?
Anyways the exchange should have an exchange zone that is the combination of both zones ?

Reply:
First we define overrides for them here in the ENTSOE parser:
```python
    "FR-COR-AC->IT-SAR": [
        ENTSOE_DOMAIN_MAPPINGS["IT-SACOAC"],
        ENTSOE_DOMAIN_MAPPINGS["IT-SAR"],
    ],
    "FR-COR-DC->IT-SAR": [
        ENTSOE_DOMAIN_MAPPINGS["IT-SACODC"],
        ENTSOE_DOMAIN_MAPPINGS["IT-SAR"],
    ],
```
Then in FR-COR_IT-SAR.py we do this:
```python
   # IT-SACOAC to IT-SAR
    AC_dataList = EXCHANGE_FUNCTION_MAP[type](
        zone_key1="FR-COR-AC",
        zone_key2="IT-SAR",
        session=session,
        target_datetime=target_datetime,
        logger=logger,
    )
    # IT-SACODC to IT-SAR
    DC_dataList = EXCHANGE_FUNCTION_MAP[type](
        zone_key1="FR-COR-DC",
        zone_key2="IT-SAR",
        session=session,
        target_datetime=target_datetime,
        logger=logger,
    )
    returnList: List[dict] = []

    # Compare the length and datetimes of the two data lists and
    # if they are the same combine the data from the two lists.
    if (
        len(AC_dataList) == len(DC_dataList)
        and AC_dataList[0]["datetime"] == DC_dataList[0]["datetime"]
        and AC_dataList[-1]["datetime"] == DC_dataList[-1]["datetime"]
    ):
        logger.info("Clean match! Merging data with zip")
        for AC, DC in zip(AC_dataList, DC_dataList):
            returnList.append(
                {
                    "datetime": AC["datetime"],
                    "netFlow": AC["netFlow"] + DC["netFlow"],
                    "sortedZoneKeys": "FR-COR->IT-SAR",
                    "source": AC["source"],
                }
            )
```
There is a fallback and a bit more code around it but that is the basics.

Would be nice if we could use the new merge function with this instead of my custom implementation.

Reply:
Ok, so I believe here the easiest would be to just add a correct mapping. I don't think it should be the merge function that should take care of this.